BASE/src/Microsoft.ApplicationInsights/Channel/Transmission.cs

```diff
@@ -125,6 +136,24 @@ public string Id
             get; private set;
         }
 
+        /// <summary>
+        /// Gets a value indicating whether flush is async.
+        /// </summary>
+        public bool ManualFlushAsyncFlag
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// Gets Test to support async Flush.
+        /// </summary>
+        public TaskCompletionSource<bool> FlushTaskCompletionSource
```

Comment:
Is this really supposed to be public? It feels like implementation detail. Should it be internal?
Lower down, there's also a public `SetFlushTaskCompletionSource` method which is probably misnamed -- suggest: "SetFlushTaskCompletionSourceResult".

There's probably a narrower API we can expose from Transmission to support manual flushing.

Reply:
Changed name to SetFlushTaskCompletionSourceResult. Transmission is a part of Microsoft.ApplicationInsights project. ServerTelemetryChannel is a part of different project. Most properties in this class are exposed as public.

Reply:
I recommend **not** exposing the FlushTaskCompletionSource. Make it a private field. Instead, expose a `FlushTask` property which returns the `.Task` of that private field.